multiboot2/src/lib.rs

```diff
@@ -262,6 +262,12 @@ impl BootInformation {
         self.get_tag::<MemoryMapTag, _>(TagType::Mmap)
     }
 
+    /// Search for the Memory map tag, return a mutable reference.
+    pub fn memory_map_tag_mut(&mut self) -> Option<&mut MemoryMapTag> {
```

Comment:
yikes, this one is controversial for me. Why do you want to change the multiboot2 boot information? I do not like the idea.. I think, a more clean approach would be to copy an owned version into your kernel, bootloader or whatever and modify the data there.

Reply:
I'm splitting this off for now. :)